### PR TITLE
committing the code for the salesforce billing and marketing crm

### PR DESCRIPTION
**Pull request by ishrartx**

### CODE REVIEW

**Review by snaphray — COMMENTED** [review]
Please check before raising the PR. Raise the PR for Epic as you have raised it for Main branch

File: src/test/java/SalesforceModules/QuoteUtil.java [hunk 1]
```diff
@@ -582,7 +582,23 @@ public static void createNewQuoteForMarketing(String name, String opportunityNam
         selectAccount(name,name+" entered account name");
         selectType(type,"selected type");
         clickSaveButton("clicked save button");
-        addDiscountProduct();
+        //addDiscountProduct();
```
Comment (snaphray) [inline src/test/java/SalesforceModules/QuoteUtil.java]: do we need this commented line?

File: src/test/java/pageobjects/SalesforceObjects/BillingObject.java [hunk 2]
```diff
@@ -29,4 +29,17 @@ public class BillingObject {
 
     public static By searchProduct = By.xpath("//input[@placeholder='Search this list...']");
 
+    public static By viewAll= By.xpath("//button[@class='slds-button' and @type='button' and text()='View All']");
+    public static By salesforceBilling= By.xpath("//p[@class='slds-truncate' and text()='Salesforce Billing']");
+    public static By accountNameLink= By.xpath("//a[@title='TXTest' and text()='TXTest']");
+    public static By NewBtnContactsOfAccount= By.xpath("//button[@class='slds-button slds-button_neutral' and @name='NewContact' and text()='New']");
```
Comment (snaphray) [inline src/test/java/pageobjects/SalesforceObjects/BillingObject.java]: It should start from small letter

**Comment by ishrartx** [comment]
Yes we need this .this will used in marketing crm for creating quote functionality

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: src/test/java/step_definitions/SalesForceMarketing.java [hunk 3]
```diff
@@ -294,4 +294,9 @@ public static void changeLeadStatus() {
             CampaignUtil.verificationMessage();
         }
     }
+
+    @And("user create Quotas an add Product in marketing")
+    public void createQuoteAndAddproduct(){
```
Comment (snaphray) [inline src/test/java/step_definitions/SalesForceMarketing.java]: why this method is empty and there is no implementation?

File: src/test/resources/features/SalesForceBilling.feature [hunk 4]
```diff
@@ -18,3 +18,18 @@ Feature: salesforce demo
...
       | TestData |
       | TestData1 |
+
+  @salesforceBill
+  Scenario Outline: salesforce demo
+    Given Read the test data  "<TestData>" from Excel
+    When Navigate to Salesforce as "BillingAnother Admin"
+    And Login to Salesforce with "BillingAnother Admin" credentials
+    And Click on Okay
+    And user clicks on the App launcher button
```
Comment (snaphray) [inline src/test/resources/features/SalesForceBilling.feature]: Please follow the same pattern throughout the feature file, start the steps from capital letter

File: src/test/resources/features/SalesMarketingCRM.feature [hunk 5]
```diff
@@ -29,18 +29,21 @@ Feature: Salesforce Marketing CRM
...
-    Then user added case in the contact
-    Then user able to see the case history
-    And user click on the lead Tab
-    Then user create new lead
-    And user click on the lead Tab
-    Then user selects the filter in Lead
-    Then user edit the lead
-    Then user delete the lead
-    Then user remove the filter
+    When Click on Opportunity Tab in converted popUp
```
Comment (snaphray) [inline src/test/resources/features/SalesMarketingCRM.feature]: check for this feature file as well

**Review by snaphray — COMMENTED** [review]
Please resolve the review comments

File: src/test/java/step_definitions/SalesForceMarketing.java [hunk 6]
```diff
@@ -82,120 +82,120 @@ public void Select_Parent_Campaign() throws InterruptedException {
...
-    @Then("user clicks on the Contact tab")
+    @Then("User clicks on the Contact tab")
     public void click_contactTab() throws InterruptedException {
         CampaignUtil.clickContactTab();
 
     }
 
-    @Then("user selects the filter")
+    @Then("User selects the filter")
     public void selectFilter() throws InterruptedException {
```
Comment (ShruthiNiranjan) [inline src/test/java/step_definitions/SalesForceMarketing.java:123]: Sometimes you are using underscore for the function names and sometimes without it. Please use same naming convention for functions.

File: src/test/java/step_definitions/SalesForceMarketing.java [hunk 7]
```diff
@@ -82,120 +82,120 @@ public void Select_Parent_Campaign() throws InterruptedException {
...
         CampaignUtil.selectFilterInContacts(dataMap.get("FieldValue"),dataMap.get("ContactLastName"));
         KeywordUtil.takeScreenshotAndAttachInReport();
     }
 
-    @Then("user edit the contact")
+    @Then("User edit the contact")
     public void edit_Contact() throws InterruptedException {
```
Comment (ShruthiNiranjan) [inline src/test/java/step_definitions/SalesForceMarketing.java]: Please change the function names with and without underscore. Make changes uniformly.

File: src/test/java/SalesforceModules/QuoteUtil.java [hunk 8]
```diff
@@ -340,34 +342,112 @@ public static void clickGenerateDocument(String logStep) throws InterruptedExcep
         goToOpportunity("navigated to opportunity");
     }
 
+    /**
+     * Enter discount and select product using JS paths
```
Comment (ShruthiNiranjan) [inline src/test/java/SalesforceModules/QuoteUtil.java]: Please remove commented lines if not required.

File: src/test/java/SalesforceModules/QuoteUtil.java [hunk 9]
```diff
@@ -582,7 +657,88 @@ public static void createNewQuoteForMarketing(String name, String opportunityNam
...
+                    HTMLReportUtil.passStringGreenColor(" <font color='green'>select product</font> page"));
+            //clicking save button using JS path
+            WebElement save = KeywordUtil.excuteJavaScriptExecutorScripts("return document.querySelector('#sbPageContainer').querySelector('#content > sb-line-editor').querySelector('#actions > sb-custom-action:nth-child(13)').querySelector('#mainButton')");
+            executor.executeScript("arguments[0].click();",save);
+            RunCukesTest.logger.log(LogStatus.PASS,
+                    HTMLReportUtil.passStringGreenColor(" <font color='green'>clicked save button</font> page"));
+            //switching back to parent frame
+            KeywordUtil.getDriver().switchTo().parentFrame();
+
+
```
Comment (ShruthiNiranjan) [inline src/test/java/SalesforceModules/QuoteUtil.java]: please remove the large space in between the lines.

**Review by snaphray — APPROVED** [review]
Code looks fine

**Pull request merged**